### PR TITLE
Added the Introduction to the Intermediate Algorithm Scripting Challenges

### LINKED ISSUE #34401: add introductory text to intermediate algorithm scripting
**Describe your problem and how to reproduce it:**
There is no introductory test to this part of the challenges, only a stub. 

**Add a Link to the page with the problem:**
https://learn.freecodecamp.org/javascript-algorithms-and-data-structures/intermediate-algorithm-scripting

**Tell us about your browser and operating system:**
* Browser Name:  Mozilla Firefox
* Browser Version: 63.0.3
* Operating System: Win10 64-bit

**If possible, add a screenshot here (you can drag and drop, png, jpg, gif, etc. in this box):**
gg

### PR DESCRIPTION
**Pull request by thecodingaviator**
- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] My article does not contain shortened URLs or affiliate links.

If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.

Closes #34401

### CODE REVIEW

File: client/src/pages/learn/javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/index.md
Comment (RandellDawson) [inline client/src/pages/learn/javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/index.md]: In American English, it is "learned" not "learnt".
```suggestion
These challenges will allow you to test how much you have learned and which parts you may need to brush up again before starting the projects.
```

File: client/src/pages/learn/javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/index.md
Comment (RandellDawson) [inline client/src/pages/learn/javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/index.md]: In case the number of challenges decreases or increase, just remove the 21.  Also, see my suggested change for the second sentence, to keep prevent so many prepositional phrases (3).
```suggestion
The following challenges are part of FCC's Intermediate Algorithm Scripting Challenges. These should prepare you to complete the final challenges, which will complete the Javascript Algorithms And Data Structures Certification.
```

**Review by RandellDawson — CHANGES_REQUESTED** [review]
@thecodingaviator See my suggestions.

File: client/src/pages/learn/javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/index.md
Comment (thecodingaviator) [reply client/src/pages/learn/javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/index.md]:
```suggestion
The following challenges are part of FCC's Intermediate Algorithm Scripting Challenges. These should prepare you to complete the final challenges for the JavaScript Algorithms And Data Structures Certification.
```
How does this sound?

Comment (RandellDawson) [reply client/src/pages/learn/javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/index.md]: I like your latest suggestion.  I tried to commit the changes, but you have not given maintainers permission to make changes to your branch for this PR.

Comment (thecodingaviator) [reply client/src/pages/learn/javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/index.md]: This box is turned on:
image

Comment (RandellDawson) [reply client/src/pages/learn/javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/index.md]: Strange.  It will not let me commit the changes.  I just tried again.

image

Can you just make the changes, so that I can review the final changes?

Comment (thecodingaviator) [reply client/src/pages/learn/javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/index.md]: Alright, one sec

**Comment by thecodingaviator** [comment]
@RandellDawson Done

**Review by RandellDawson — APPROVED** [review]

**Comment by RandellDawson** [comment]
@thecodingaviator I am fine with the changes.  @cmccormack What do you think?

File: client/src/pages/learn/javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/index.md
Comment (cmccormack) [inline client/src/pages/learn/javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/index.md]: is to "bush up" a common enough idiom to be used here?  Can we changes this to something more clear to non-native english speakers, maybe just "review again"?

Comment (thecodingaviator) [reply client/src/pages/learn/javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/index.md]: Let's get @RandellDawson 's opinion. I'm fine with either

Comment (RandellDawson) [reply client/src/pages/learn/javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/index.md]: That is fine.

Comment (thecodingaviator) [reply client/src/pages/learn/javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/index.md]: @cmccormack You want me to change it?

Comment (RandellDawson) [reply client/src/pages/learn/javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/index.md]: I would go ahead and implement @cmccormack's suggestion.  That is the only thing holding up this PR from being merged.

Comment (thecodingaviator) [reply client/src/pages/learn/javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/index.md]: Going ahead

Comment (thecodingaviator) [reply client/src/pages/learn/javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/index.md]: @RandellDawson Fixed.

**Review by cmccormack — APPROVED** [review]
This looks good to me 👍.  Thank you for your contribution!

**Pull request merged**
